src/mds/MDCache.cc

```diff
@@ -10090,8 +10090,9 @@ void MDCache::request_cleanup(const MDRequestRef& mdr)
     auto new_batch_head = it->second->find_new_head();
     if (!new_batch_head) {
       mdr->batch_op_map->erase(it);
+    } else {
+      mds->finisher->queue(new C_MDS_RetryRequest(this, new_batch_head));
```

Comment:
this looks sensible to me ... wrapping the re-queuing in an `else` block probably got missed earlier